data_specification/enums/data_type.py

```diff
@@ -311,6 +363,8 @@ def __init__(self, value, size, min_val, max_val, scale, struct_encoding,
         self.scale = scale
         self.struct_encoding = struct_encoding
         self.numpy_typename = numpy_typename
+        self.numpy_decoding = numpy_decoding
+        self.numpy_decoding_size = nump_decoding_size
```

Comment:
These two things are only ever used together (and that would have to be true everywhere).  Why not just have a single variable for this?

Reply:
Actually, I also just realised that there is already a numpy dtype.  Does that not work?

Reply:
think this is related to the neuron recorder stuff. so this may be a dead pr again. 

to the question. the numpy dtype and numpy decoding terms do not match in all cases. annoyingly. 

why have 2 instead of together. clearer. you could argue the same for a lot of those variables. but they're all separate.